Rails実践課題②（宿泊予約アプリ）を作成

ezulv37 opened the pull request:
### 実装内容

＜共通＞
・rootメソッドでルート（/）へのルーティング設定
・チェックイン/チェックアウト/施設作成日を、yyyy/mm/ddの形式で表示
・予約確定日時を、yyyy/mm/dd HH:MMの形式で表示
・ 施設画像やアイコン画像が登録されていない場合、デフォルト画像を表示

＜ユーザー関連＞
●ユーザー登録/編集機能を実装
・新規登録は、名前、メールアドレス、パスワード、確認用パスワードを必須項目
・アカウントページを作成し、メールアドレス、パスワードを表示
・パスワードは*********
・アカウント情報（メールアドレス、パスワード）の編集ページを作成し、編集機能を実装
・プロフィール（アイコン画像、名前、自己紹介）の編集ページを作成し、編集機能を実装
●ログイン/ログアウト機能を実装
・メールアドレスとパスワードでログイン可能
・ログインの有無により、表示項目を切り替え
　　【ログイン中の場合】
 　　　・ヘッダーにアイコン画像、ユーザー名を表示
 　　　・ユーザー名をクリックするとドロップダウンメニューが表示
　　【ログアウト中の場合】
 　　　・ヘッダーに「ログイン」「新規登録」のリンクを表示

＜施設関連＞
●施設一覧/詳細ページおよび施設の作成機能を実装
・施設一覧ページには、自分が作成したデータを対象とし、施設画像、施設名、紹介文、宿泊料金（日）、作成日を表示
・施設名を施設詳細ページへのリンクとして実装
・施設登録は、施設名、施設詳細、料金、住所を必須項目
・料金が1円以上となるよう、バリデーションを設定
・施設詳細ページには、施設画像、施設名、紹介文、宿泊料金（/日）の情報を表示
・施設詳細ページには、直前のページへ戻るためのリンクを実装
●施設情報の検索機能を実装
・検索結果ページには、施設の合計件数、施設情報（施設名、住所、施設詳細、宿泊料金（/日））を表示
・施設情報には、施設詳細ページへのリンクを配置
　　【エリア検索機能について】
　　　・エリアは、東京・大阪・京都・札幌の4つ
　　　・住所を対象とした「あいまい検索」
　　【フリーワード検索機能について】
　　　・施設名・施設詳細を対象とした「あいまい検索」


＜予約関連＞
●施設の予約機能を実装
・施設予約は、チェックイン日、チェックアウト日、人数を必須項目
・チェックイン日が本日以降の日付となるよう、バリデーションを設定
・チェックアウト日がチェックイン日より後の日付となるよう、バリデーションを設定
・ 人数が1人以上となるよう、バリデーションを設定
・予約確認ページで、チェックイン日、チェックアウト日、宿泊日数、人数、支払い金額を表示
・支払い金額を、宿泊日数、人数を加味した合計金額で表示
●予約した施設の一覧ページを作成
・自分が予約した施設データのみ表示
・一覧ページに、施設画像、施設名、紹介文、合計料金、チェックイン日、チェックアウト日、予約確定日時を表示

Commit:
ParameterMissing (param is missing or the value is empty)のエラーが解消できず、予約確認画面から予約の確定ができない

Commit:
params[:id].present?がfalseになり予約の更新ができない

Commit:
宿泊予約アプリの完成

T-Hinano commented:
bundle installをするとrmagickのインストールでエラーが発生します。
git cloneで別のディレクトリにこのリポジトリをクローンしてbundle installを実行してみて下さい。
レビュワーは下記の手順でRailsを立ち上げています。

git clone <リポジトリURL>
cd <クローンしたプロジェクトのディレクトリ名>
git checkout -b develop
git pull origin develop
bundle install
rails db:migrate
rails webpacker:install
rails s

ローカルで同様のエラーが発生するか確認して、再現した場合は修正して下さい。
もしローカルでエラーが発生しなかった場合はお手数ですがその旨をコメントに追記して再提出して下さい。

ezulv37 commented:
レビューありがとうございます。
かしこまりました。こちらのローカル環境ではエラーが発生しませんでしたので再提出させていただきます。

T-Hinano commented:
ご確認ありがとうございます。
こちらで確認したところrmagickだけbundle installでインストールできなかったので、一旦Gemfileから外し別の手順でインストールして対応しました。

その後rails db:migrateを実行したところ、画像のエラーが発生しました。 こちらの問題を修正して再度提出をお願いします。
<img width="608" alt="スクリーンショット 2025-05-08 12 41 54" src="https://github.com/user-attachments/assets/a0ac482b-766d-4bed-b5bb-cbe245adec2f" />

ローカル環境でrails db:migrate:resetを実行して、エラーを確認後、 マイグレーションファイルを修正（不要であれば削除）してください。
修正後は再度rails db:migrate:resetを実行を繰り返してマイグレーションファイルに問題がないことを確認してください。
このコマンドによりDBに登録したデータは一度全て削除されますのでご注意ください。
※チーム開発であれば、一度プッシュしたマイグレーションファイルの編集はNGとなりますが、今回は一人での開発となりますので編集いただいて大丈夫です。

ezulv37 commented:
ありがとうございます。かしこまりました。

ターミナルで　rm db/migrate/20250502113121_add_number_of_date_to_reservations.rb　を実行し、マイグレーションファイルを１つ削除しました。
その後、db:migrate:reset　を実行しました。エラー文が出ませんでしたので、再提出させていただきます。

```
% rails db:migrate:reset
Running via Spring preloader in process 6660
Dropped database 'db/development.sqlite3'
Dropped database 'db/test.sqlite3'
Created database 'db/development.sqlite3'
Created database 'db/test.sqlite3'
== 20250428092650 DeviseCreateUsers: migrating ================================
-- create_table(:users)
   -> 0.0008s
-- add_index(:users, :email, {:unique=>true})
   -> 0.0004s
-- add_index(:users, :reset_password_token, {:unique=>true})
   -> 0.0003s
== 20250428092650 DeviseCreateUsers: migrated (0.0014s) =======================

== 20250428093752 AddColumnsToUsers: migrating ================================
-- add_column(:users, :image, :string)
   -> 0.0007s
-- add_column(:users, :name, :string)
   -> 0.0003s
-- add_column(:users, :introduction, :string)
   -> 0.0004s
== 20250428093752 AddColumnsToUsers: migrated (0.0014s) =======================

== 20250430084338 CreateActiveStorageTables: migrating ========================
-- create_table(:active_storage_blobs, {:id=>:primary_key})
   -> 0.0063s
-- create_table(:active_storage_attachments, {:id=>:primary_key})
   -> 0.0010s
-- create_table(:active_storage_variant_records, {:id=>:primary_key})
   -> 0.0006s
== 20250430084338 CreateActiveStorageTables: migrated (0.0079s) ===============

== 20250501023434 CreateRooms: migrating ======================================
-- create_table(:rooms)
   -> 0.0007s
== 20250501023434 CreateRooms: migrated (0.0007s) =============================

== 20250501050100 AddColumnsToRooms: migrating ================================
-- add_column(:rooms, :address, :string)
   -> 0.0006s
== 20250501050100 AddColumnsToRooms: migrated (0.0006s) =======================

== 20250502022559 CreateReservations: migrating ===============================
-- create_table(:reservations)
   -> 0.0009s
== 20250502022559 CreateReservations: migrated (0.0009s) ======================

== 20250502031151 AddColumnsToReservations: migrating =========================
-- add_column(:reservations, :number_of_guests, :integer)
   -> 0.0006s
-- add_column(:reservations, :number_of_date, :integer)
   -> 0.0004s
== 20250502031151 AddColumnsToReservations: migrated (0.0010s) ================

== 20250502043019 AddRoomIdToReservations: migrating ==========================
-- add_column(:reservations, :room_id, :integer)
   -> 0.0006s
== 20250502043019 AddRoomIdToReservations: migrated (0.0006s) =================

== 20250505055243 AddUserToRooms: migrating ===================================
-- add_reference(:rooms, :user, {:null=>false, :foreign_key=>true})
   -> 0.0101s
== 20250505055243 AddUserToRooms: migrated (0.0101s) ==========================

== 20250505062654 AddUserToReservations: migrating ============================
-- add_reference(:reservations, :user, {:null=>false, :foreign_key=>true})
   -> 0.0380s
== 20250505062654 AddUserToReservations: migrated (0.0380s) ===================

== 20250505081351 AddAreaToRooms: migrating ===================================
-- add_column(:rooms, :area, :string)
   -> 0.0008s
== 20250505081351 AddAreaToRooms: migrated (0.0010s) ==========================
```

Commit:
rm db/migrate/20250502113121_add_number_of_date_to_reservations.rb　を実行し、マイグレーションファイルを１つ削除。
その後、db:migrate:reset　を実行。エラー文なしのため再提出

T-Hinano commented on app/assets/stylesheets/application.css:
```diff
@@ -13,3 +13,86 @@
+  padding: 15px;
+  height: 150px;
+  background-color: #1c1d3b;
+}
```
この赤いマークはファイルの最後に改行がないか、改行コードに不備がある場合に表示されます！
今回は修正不要ですが、環境によっては不具合の原因になることもあるので表示されないように調整しましょう！

T-Hinano commented on app/assets/stylesheets/confirms.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the confirms controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
提出時点でコメントのみのSCSSファイルは削除して大丈夫です！

T-Hinano commented on app/assets/stylesheets/homes.scss:
```diff
@@ -1,3 +1,133 @@
 // Place all the styles related to the Homes controller here.
 // They will automatically be included in application.css.
 // You can use Sass (SCSS) here: https://sass-lang.com/
```
記述の補足ではないコメントは削除して大丈夫です！

T-Hinano commented on app/assets/stylesheets/rooms.scss:
```diff
@@ -0,0 +1,269 @@
+
+
+
+
```
不要な改行はなるべく避けて1~2行までにするようにしましょう！

T-Hinano reviewed (approved):
LGTMです！
全体的に不要な改行が多く残っていた印象でした。
プルリクエストを作成した後は差分のタグからセルフコードレビューする工程を行いましょう！
綺麗にサイトを作ることができていても、改行が多いとソースコードが雑に見えてしまいます。次から意識して整えていきましょう。

ソースコードがより読みやすくなるようにいくつかコメントでアドバイスさせていただきました！
今後の参考にしていただけると幸いです！

ezulv37 commented:
レビューいただきありがとうございました。
PR後のセルフレビューを意識していきます。ありがとうございました。
